### PR TITLE
Interpolate ki/kp for steering PID loop

### PR DESCRIPTION
**Pull request by vntarasov**
Very much needed for the Volt port: car ping-pongs with low kp on high speeds, and the loop is unstable with high kp on low speeds.

Also, removes "number or array?" logic from PIController, now that all the callers use interpolation for ki/kp.

### CODE REVIEW

File: cereal/car.capnp [hunk 1]
```diff
@@ -301,8 +301,12 @@ struct CarParams {
   tireStiffnessRear @14 :Float32;    # [N/rad] rear tire coeff of stiff
 
   # Kp and Ki for the lateral control
-  steerKp @15 :Float32;
-  steerKi @16 :Float32;
+  steerKpBP @42 :List(Float32);
+  steerKpV @43 :List(Float32);
+  steerKiBP @44 :List(Float32);
+  steerKiV @45 :List(Float32);
+  steerKpDEPRECATED @15 :Float32;
```
Comment (rbiasini) [inline cereal/car.capnp]: @vntarasov , since carParams isn't logged, you can remove the DEPRECATED ones.

**Comment by rbiasini** [comment]
@vntarasov this is a good generalization (we are already doing it for long control). I made a minor comment, other than that it looks ok.

File: cereal/car.capnp (hunk 1 above)
Comment (vntarasov) [reply cereal/car.capnp]: Thanks! Should I then clean up existing DEPRECATED carParams fields as well?

Comment (rbiasini) [reply cereal/car.capnp]: No worries, I'll do the clean up on our internal branch, since it will generate a lot of index shifting.

**Comment by mwkrieger** [comment]
Will this help with ping-ponging in general, or just for the Volt? I'm seeing high rates of ping-pong at higher speeds >~60mph.

**Comment by vntarasov** [comment]
@mwkrieger I don't have data for other cars. But I do believe openpilot needs a self-tuning PID loop for steering, parameterized by speed.

**Comment by fiftycentsjj** [comment]
@rbiasini , Greetings to you! Will I need to compile the file again, after modifying car.capnp? Thanks!

**Comment by rbiasini** [comment]
@vntarasov , one last change so that the Python linter check will not fail. Can you fix this?
```
Pyflakes found errors in the code. Please fix and try again
selfdrive/controls/lib/pid.py:3: 'numbers' imported but unused
Exited with code 1
```

I've already patched this PR in our internal repo, so after this change I can go ahead and merge it.

**Comment by vntarasov** [comment]
@rbiasini, done, thanks!

**Pull request merged**